pkg/artifacts/signable.go

```diff
@@ -257,6 +276,59 @@ func extractTargetFromResults(obj objects.TektonObject, identifierSuffix string,
 	return ss
 }
 
+// ExtractStructuredTargetFromResults extracts structured signable targets aim to generate intoto provenance as materials within TaskRun results and store them as StructuredSignable.
+func ExtractStructuredTargetFromResults(obj objects.TektonObject, categoryMarker string, logger *zap.SugaredLogger) []*StructuredSignable {
```

Comment:
it wasn't obvious to me what categoryMarker meant just by reading the header + godoc. Perhaps rename (nameSuffix?) and/or add more godoc to explain?

Reply:
I've added godoc since I'm trying to make it clear that the type hinting is also distinguishing artifacts into inputs and outputs

Reply:
Good point! Added a check there